### PR TITLE
wallet-ext: qredo signer

### PR DESCRIPTION
**Pull request by pchrysochoidis**
* allow signing and executing transactions with qredo account
* shows a notification dialog while waiting for custody approval of transaction

https://github.com/MystenLabs/sui/assets/10210143/04047020-ffc8-4640-bb0d-9b3f94c10422

https://github.com/MystenLabs/sui/assets/10210143/0aea3dc4-48e0-445a-a963-9dc64f9695ad

https://github.com/MystenLabs/sui/assets/10210143/395d8697-886f-466e-bb79-ae827d561ae1

https://github.com/MystenLabs/sui/assets/10210143/ed5352d2-e7d9-47da-8092-172de152797c

closes APPS-634

### CODE REVIEW

**Comment by pchrysochoidis** [comment]
> ⚠️ 🦋 **Changesets Warning:** This PR has changes to public npm packages, but does not contain a changeset. You can create a changeset easily by running `pnpm changeset` in the root of the Sui repo, and following the prompts. If your change does not need a changeset (e.g. a documentation-only change), you can ignore this message. This warning will be removed when a changeset is added to this pull request.
> 
> [Learn more about Changesets](https://github.com/changesets/changesets).

just extracted some code to a function to reuse it - no need for changeset

**Comment by mystie711** [comment]
image
Added the spinner location in Figma. Please use that.

**Review by mystie711 — APPROVED** [review]
Looks good once the progress spinner location is adjusted along with the Qredo icon in the modal.

File: apps/wallet/src/ui/app/hooks/useQredoTransaction.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import { ModalDialog } from '../shared/ModalDialog';
+import { Heading } from '../shared/heading';
+import { Text } from '../shared/text';
+
+export function useQredoTransaction() {
+    const [clientIdentifier, setClientIdentifier] = useState(() => uuidV4());
+    const [qredoTransactionID, setQredoTransactionID] = useState<string | null>(
+        null
+    );
+    const notificationModal = (
```
[the remark this answers is not in the document]
Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/hooks/useQredoTransaction.tsx:19]: Generally a toast will be shown. Any specific reason you asking about it? The dialog should always hide itself.

File: apps/wallet/src/ui/app/hooks/useQredoTransaction.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        QredoEvents.on('qredoTransactionCreated', qredoTransactionCreated);
+        QredoEvents.on('qredoActionDone', qredoTransactionComplete);
+        return () => {
+            QredoEvents.off('qredoTransactionCreated', qredoTransactionCreated);
+            QredoEvents.off('qredoActionDone', qredoTransactionComplete);
+            QredoEvents.emit('clientIgnoredUpdates', { clientIdentifier });
+        };
+    }, [clientIdentifier]);
+    return {
+        clientIdentifier,
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/hooks/useQredoTransaction.tsx:80]: What if you send two qredo transactions during a session, could that possible result int he same client identifier?

**Pull request merged**